test/MOI/MOI_callbacks.jl

```diff
@@ -82,9 +82,8 @@ function test_lazy_constraint_callback()
                 model,
                 MOI.CallbackNodeStatus(cb_data),
             )::MOI.CallbackNodeStatusCode
-            if round.(Int, [x_val, y_val]) ≈ [x_val, y_val]
-                atol = 1e-6
-                @test status == MOI.CALLBACK_NODE_STATUS_INTEGER
+            if status == MOI.CALLBACK_NODE_STATUS_INTEGER
+                @test isapprox(round.(Int, [x_val, y_val]), [x_val, y_val], atol = 1e-5)
```

Comment:
Ah. This looks like a formatting error from JuliaFormatter where the `atol` got dropped onto the wrong line.

The test should be
```Julia
               int_sol = round.(Int, [x_val, y_val])
               if isapprox(int_sol, [x_val, y_val], atol = 1e-6)
                 @test status == MOI.CALLBACK_NODE_STATUS_INTEGER
```

This revision isn't correct, because it's testing the reverse implication of what we want to test.

Reply:
I'll sort the edit

Reply:
I changed the test because it seems like it would be numerically sensitive

Reply:
it failed again. is there a reason it is testing the implication that way and not the other way?

Reply:
We want to check that if you are at an integer solution, Gurobi will report that the `CallbackNodeStatus` is `INTEGER`. The reverse implication is that if the node status is integer, then the solution is integral. But we might never hit the `INTEGER` status in our tests. 

This seems weird that we have an integer solution, but Gurobi is claiming it is at a MIPNODE not a MIPSOL. Perhaps I need to tighten the tolerances.

It's also weird that this was passing tests prior. But perhaps that's because of Julia 1.7 changing the random numbers, rather than a complication of your PR.